libs/design-system/src/atoms/progress/progressBar.tsx

```diff
@@ -0,0 +1,30 @@
+import './../../index.css';
+
+import { cn } from '../../shared/classnames';
+
+type ProgressBarProps = {
+  /**
+   * The ratio of the progress bar. This should be a number between 1 and 100.
+   */
+  ratio: number;
+
+  /**
+   * The className to apply to the text. This will override any built-in style.
+   *
+   * this is useful for layout positioning, responsive sizing, and for any stylistic
+   * exception that is not part of our design system (eg. a specific size used only once)
+   */
+  className?: string;
+};
+
+const percentagesMapping = Array(101)
+  .fill(0)
+  .map((_, i) => `plasmo-w-[${i}%]`);
```

Comment:
This doesn't work because tailwind classes are generated at the build and not after. I'll make another suggestion